source/Test.cpp

```diff
@@ -537,6 +559,25 @@ void Test::Fail(const Context &context, const PlayerInfo &player, const string &
 	
 	if(!testFailReason.empty())
 		message += ": " + testFailReason;
+
+	// Print some debug information about the flagship and the first 5 escorts.
+	const Ship *flagship = player.Flagship();
+	if(!flagship)
+		Files::LogError("Player did not have flagship at the moment of failure.");
+	else
+	{
+		Files::LogError("flagship: " + ShipToString(*flagship));
+		int escortsPrinted = 0;
+		for(auto escortWeakPtr: flagship->GetEscorts())
+		{
+			auto escort = escortWeakPtr.lock();
+			if(!escort)
+				continue;
+			Files::LogError("escort: " + ShipToString(*(escort.get())));
+			if(escortsPrinted >= 5)
```

Comment:
Might be a good idea to add a message saying that there are X more escorts not displayed. Also you never increment that variable.